### PR TITLE
Test fix: normalized filepaths before comparison

### LINKED ISSUE #10132: TST: test_data_name_third_party_package fails when OS is case-insensitive
I noticed this when I ran remote data tests on Windows and my path had mixed case. Not sure if it is worth fixing. 🤷 
```
_____________________ test_data_name_third_party_package ______________________

    def test_data_name_third_party_package():
        """Regression test for issue #1256

        Tests that `get_pkg_data_filename` works in a third-party package that
        doesn't make any relative imports from the module it's used from.

        Uses a test package under ``data/test_package``.
        """

        # Get the actual data dir:
        data_dir = os.path.join(os.path.dirname(__file__), "data")

        sys.path.insert(0, data_dir)
        try:
            import test_package

            filename = test_package.get_data_filename()
>           assert filename == os.path.join(data_dir, "test_package", "data", "foo.txt")
E           AssertionError: assert 'D:\\...\\S...data\\foo.txt' == 'd:\\...\\st...data\\foo.txt'
E             - D:\...\STScI\ssb\astropy\astropy\utils\tests\data\test_package\data\foo.txt
E             ? ^      ^^^ ^
E             + d:\...\stsci\ssb\astropy\astropy\utils\tests\data\test_package\data\foo.txt
E             ? ^      ^^^ ^

astropy\utils\tests\test_data.py:888: AssertionError
```

### PR DESCRIPTION
**Pull request by vdye**
### Description

This pull request addresses a Windows-specific issue where a remote data test fails due to path string mismatches (paths aren't case-sensitive on Windows). 

Fixes #10132

### CODE REVIEW

**Comment by pllim** [comment]
Thanks! Seems fine but for some reason, not all checks kicked off, so I am going to close and re-open this again to try to kick them all off.

**Comment by pllim** [comment]
Hmm, really want to see that Travis CI but it is not kicking off. Maybe server-side issues. I'll wait a bit and come back to this.

**Comment by vdye** [comment]
@pllim It looks like the Travis CI pipeline ran, but an (I think) unrelated test failed?

File: astropy/utils/tests/test_data.py [hunk 1]
```diff
@@ -885,7 +885,8 @@ def test_data_name_third_party_package():
         import test_package
 
         filename = test_package.get_data_filename()
-        assert filename == os.path.join(data_dir, "test_package", "data", "foo.txt")
+        expected_filename = os.path.join(data_dir, "test_package", "data", "foo.txt")
+        assert os.path.normcase(filename) == os.path.normcase(expected_filename)
```
[the remark this answers is not in the document]
Comment (vdye) [reply astropy/utils/tests/test_data.py]: Updated!

**Comment by bsipocz** [comment]
This has been already released, but milestone hasn't been updated accordingly, fixing it now.

**Pull request merged**